include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -753,8 +753,25 @@ DEAL_II_NAMESPACE_OPEN // Do not convert for module purposes
      * prior to use (in the compress() step).
      */
     SparseMatrix(const IndexSet    &parallel_partitioning,
-                 const MPI_Comm     communicator          = MPI_COMM_WORLD,
-                 const unsigned int n_max_entries_per_row = 0);
+                 const MPI_Comm     communicator,
+                 const unsigned int n_max_entries_per_row);
+
+    DEAL_II_DEPRECATED_EARLY_WITH_COMMENT(
+      "Use the overload specifying the number of entries per row!")
+    SparseMatrix(const IndexSet &parallel_partitioning,
+                 const MPI_Comm  communicator)
+      : SparseMatrix(parallel_partitioning, communicator, 0)
+    {
+      std::abort();
+    }
+
+    DEAL_II_DEPRECATED_EARLY_WITH_COMMENT(
+      "Use the overload specifying the MPI communicator and the number of entries per row!")
+    SparseMatrix(const IndexSet &parallel_partitioning)
+      : SparseMatrix(parallel_partitioning, MPI_COMM_WORLD, 0)
+    {
+      std::abort();
+    }
```

Comment:
Would you mind replacing the calls to `std::abort()` by `Assert (false, ExcMessage("something"));`?